### PR TITLE
Edit User details

### PR DESCRIPTION
**Pull request by matt-weiner**

### CODE REVIEW

File: src/users/users.service.ts [hunk 1]
```diff
@@ -361,4 +362,20 @@ export class UsersService {
...
+     * 
+     * @param empID ID of the logged in user
+     * @param newUser new User object to update old user
+     * @returns the new User object which was used to update the user
+     */
+    async editUserDetails(empID: number, newUser: Users){
+        if(empID !== newUser.employeeId){
+            throw new UnauthorizedException();
+        }
+        await this.usersRepository.save(newUser);
```
Comment (dima618) [inline src/users/users.service.ts]: We shouldn't allow the user to change their own role, companyId, employeeId, manager status, manager, start date, position title, recognition stats, created date, updated date, deleted date.

**Review by dima618 — CHANGES_REQUESTED** [review]

File: src/users/users.service.ts [hunk 2]
```diff
@@ -261,4 +262,27 @@ export class UsersService {
...
+        }
+        const user = await this.usersRepository.findOne({employeeId, companyId: requester.companyId});
+        user.firstName = newUser.firstName;
+        user.lastName = newUser.lastName;
+        if (requester.role === Role.Admin) {
+            user.positionTitle = newUser.positionTitle;
+            user.startDate = newUser.startDate;
+            user.isManager = newUser.isManager
+        }
+        return newUser;
```
Comment (dima618) [inline src/users/users.service.ts]: Are you saving the new user details to the database? Does typeorm do that automatically?

**Review by dima618 — CHANGES_REQUESTED** [review]

File: src/users/users.service.ts (hunk 2 above)
Comment (smalhotra3599) [reply src/users/users.service.ts]: uhh no I'm not apparently lol, must've not got pushed up. One sec

Comment (smalhotra3599) [reply src/users/users.service.ts]: should be good now, my bad

**Review by dima618 — APPROVED** [review]

**Pull request merged**